diff --git a/cmake/OpenCVDetectCUDALanguage.cmake b/cmake/OpenCVDetectCUDALanguage.cmake
index bc40134180..f580b69fa5 100644
--- a/cmake/OpenCVDetectCUDALanguage.cmake
+++ b/cmake/OpenCVDetectCUDALanguage.cmake
@@ -33,12 +33,16 @@ if(CMAKE_CUDA_COMPILER AND CUDAToolkit_FOUND)
   set(CUDA_TOOLKIT_INCLUDE ${CUDAToolkit_INCLUDE_DIRS})
   set(CUDA_VERSION_STRING ${CUDAToolkit_VERSION})
   set(CUDA_VERSION ${CUDAToolkit_VERSION})
-  if(CUDA_VERSION VERSION_LESS 11.0)
-      set(CMAKE_CUDA_STANDARD 11)
-  elseif(CUDA_VERSION VERSION_LESS 12.8)
-      set(CMAKE_CUDA_STANDARD 14)
+  if(NOT CMAKE_CUDA_STANDARD)
+    if(CUDA_VERSION VERSION_LESS 11.0)
+        set(CMAKE_CUDA_STANDARD 11)
+    elseif(CUDA_VERSION VERSION_LESS 12.8)
+        set(CMAKE_CUDA_STANDARD 14)
+    else()
+        set(CMAKE_CUDA_STANDARD 17)
+    endif()
   else()
-      set(CMAKE_CUDA_STANDARD 17)
+    message(WARNING "CMAKE_CUDA_STANDARD is already set to ${CMAKE_CUDA_STANDARD}")
   endif()
   if(UNIX AND NOT BUILD_SHARED_LIBS)
       set(CUDA_LIB_EXT "_static")
@@ -153,4 +157,4 @@ endmacro()
 if(HAVE_CUDA)
   ocv_apply_cuda_stub_workaround("${CUDA_cuda_driver_LIBRARY}")
   ocv_check_cuda_delayed_load("${cuda_toolkit_root_dir}")
-endif()
\ No newline at end of file
+endif()
diff --git a/cmake/OpenCVDetectCXXCompiler.cmake b/cmake/OpenCVDetectCXXCompiler.cmake
index a8a6966108..e7fea567ab 100644
--- a/cmake/OpenCVDetectCXXCompiler.cmake
+++ b/cmake/OpenCVDetectCXXCompiler.cmake
@@ -184,6 +184,8 @@ elseif(MSVC)
     set(OpenCV_RUNTIME vc16)
   elseif(MSVC_VERSION MATCHES "^19[34][0-9]$")
     set(OpenCV_RUNTIME vc17)
+  elseif(MSVC_VERSION MATCHES "^195[0-9]$")
+    set(OpenCV_RUNTIME vc18)
   else()
     message(WARNING "OpenCV does not recognize MSVC_VERSION \"${MSVC_VERSION}\". Cannot set OpenCV_RUNTIME")
   endif()
